sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/transform/BeamBuiltinAggregations.java

```diff
@@ -293,4 +321,262 @@ public BigDecimal toBigDecimal(BigDecimal record) {
+    }
+
+    @Override
+    public KV<BigDecimal, VarAgg> addInput(KV<BigDecimal, VarAgg> accumulator, T input) {
+      double v;
+      if (input == null) {
+        return accumulator;
+      } else {
+        v = new BigDecimal(input.toString()).doubleValue();
+        accumulator.getValue().count++;
+        accumulator.getValue().sum += v;
+        double variance;
+        if (accumulator.getValue().count > 1) {
+          double t = accumulator.getValue().count * v - accumulator.getValue().sum;
+          variance = (t * t) / (accumulator.getValue().count * (accumulator.getValue().count - 1));
+        } else {
+          variance = 0;
+        }
+       return KV.of(accumulator.getKey().add(new BigDecimal(variance)), accumulator.getValue());
+      }
+    }
+
+    @Override
+    public KV<BigDecimal, VarAgg> mergeAccumulators(
+            Iterable<KV<BigDecimal, VarAgg>> accumulators) {
+      BigDecimal variance = new BigDecimal(0);
+      long count = 0;
+      double sum = 0;
+
+      Iterator<KV<BigDecimal, VarAgg>> ite = accumulators.iterator();
+      while (ite.hasNext()) {
+        KV<BigDecimal, VarAgg> r = ite.next();
+
+        double b = r.getValue().sum;
+
+        count += r.getValue().count;
+        sum += b;
+
+        double t = (r.getValue().count / (double) count) * sum - b;
+        double d = t * t
+                * ((count / (double) r.getValue().count) / ((double) count + r.getValue().count));
+        variance = variance.add(r.getKey().add(new BigDecimal(d)));
+      }
+
+      return KV.of(variance, new VarAgg(count, sum));
+    }
+
+    @Override
+    public Coder<KV<BigDecimal, VarAgg>> getAccumulatorCoder(CoderRegistry registry,
+        Coder<T> inputCoder) throws CannotProvideCoderException {
+      return KvCoder.of(BigDecimalCoder.of(), SerializableCoder.of(VarAgg.class));
+    }
+
+    public abstract T extractOutput(KV<BigDecimal, VarAgg> accumulator);
+    public abstract BigDecimal toBigDecimal(T record);
+  }
+
+  static class IntegerVar extends VarPop<Integer> {
+    boolean isSamp;
+
+    public IntegerVar(boolean isSamp) {
+      this.isSamp = isSamp;
+    }
+
+    public Integer extractOutput(KV<BigDecimal, VarAgg> accumulator) {
+      BigDecimal decimalVar = null;
+      if (accumulator.getValue().count > 1) {
+        decimalVar = accumulator.getKey().divide(
+                new BigDecimal(accumulator.getValue().count - (this.isSamp ? 1 : 0)),
+                RoundingMode.HALF_UP);
+      } else {
+        decimalVar = new BigDecimal(0);
+      }
+
+      return decimalVar.intValue();
+    }
+
+    @Override
+    public BigDecimal toBigDecimal(Integer record) {
+      return new BigDecimal(record);
+    }
+  }
+
+  static class ShortVar extends VarPop<Short> {
+    boolean isSamp;
+
+    public ShortVar(boolean isSamp) {
+      this.isSamp = isSamp;
+    }
+
+    public Short extractOutput(KV<BigDecimal, VarAgg> accumulator) {
+      BigDecimal decimalVar = null;
```

Comment:
It does. But, those are with different types. How to share same logic within `extractOutput(...)`?